mne/preprocessing/ieeg/_projection.py

```diff
@@ -100,3 +101,80 @@ def project_sensors_onto_brain(info, trans, subject, subjects_dir=None,
     for idx, loc in zip(picks_idx, proj_locs):
         info['chs'][idx]['loc'][:3] = loc
     return info
+
+
+@verbose
+def project_sensors_onto_inflated(info, trans, subject, subjects_dir=None,
+                                  picks=None, max_dist=0.006, copy=True,
+                                  verbose=None):
+    """Project sensors onto the brain surface.
+
+    Parameters
+    ----------
+    %(info_not_none)s
+    %(trans_not_none)s
+    %(subject)s
+    %(subjects_dir)s
+    %(picks_base)s only ``seeg`` channels.
+    max_dist : float
+        The maximum distance to project a sensor in meters. Sensors that
+        are greater than this distance from the pial surface will not
+        be assigned locations.
+    copy : bool
+        If ``True``, return a new instance of ``info``, if ``False``
+        ``info`` is modified in place.
+    %(verbose)s
+
+    Returns
+    -------
+    %(info_not_none)s
+
+    Notes
+    -----
+    This is useful in sEEG analysis for visualization
+    """
+    from scipy.spatial.distance import cdist
+    _validate_type(copy, bool, 'copy')
+    if copy:
+        info = info.copy()
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    surf_data = dict(lh=dict(), rh=dict())
+    x_dir = np.array([1., 0., 0.])
+    for hemi in ('lh', 'rh'):
+        for surf in ('pial', 'inflated'):
+            for img in ('', '.T1', '.T2', ''):
+                surf_fname = op.join(subjects_dir, subject, 'surf',
+                                     f'{hemi}.{surf}')
+                if op.isfile(surf_fname):
+                    break
+            coords, faces = read_surface(surf_fname)
+            if surf == 'inflated':
+                x_ = coords @ x_dir
+                coords -= np.max(x_) * x_dir if hemi == 'lh' else \
+                    np.min(x_) * x_dir
+            surf_data[hemi][surf] = (coords / 1000, faces)  # mm -> m
+    # get channel locations
+    picks_idx = _picks_to_idx(info, 'seeg' if picks is None else picks)
+    locs = np.array([info['chs'][idx]['loc'][:3] for idx in picks_idx])
+    trans = _ensure_trans(trans, 'head', 'mri')
+    locs = apply_trans(trans, locs)
+    # initialize projected locs
+    proj_locs = np.zeros(locs.shape) * np.nan
+    for hemi in ('lh', 'rh'):
+        hemi_picks = np.where(
+            locs[:, 0] <= 0 if hemi == 'lh' else locs[:, 0] > 0)[0]
+        # compute distances to pial vertices
+        dists = cdist(surf_data[hemi]['pial'][0], locs[hemi_picks])
+        for j, idx in enumerate(hemi_picks):
+            if dists[:, j].min() / 1000 < max_dist:
+                proj_locs[idx] = \
+                    surf_data[hemi]['inflated'][0][np.argmin(dists[:, j])]
```

Comment:
@alexrockhill did you consider the _DistanceQuery object and the cKDTree method?
is the computation of `dists` fast enough?

Reply:
It's very fast so I don't think it's necessary but that would be an improvement. Happy to implement that if you'd prefer.